src/scala/co/actioniq/luna/dao/SoftDeleteFilter.scala

```diff
@@ -0,0 +1,49 @@
+package co.actioniq.luna.dao
+
+import slick.jdbc.JdbcProfile
+
+import scala.concurrent.Future
+
+trait SoftDeleteFilter[T <: DAOTable.Table[V, I, P], V <: IdModel[I], I <: IdType, P <: JdbcProfile, D] extends DefaultFilter[T, V, I, P] {
+  protected val profile: JdbcProfile
+  import profile.api._
+  def softNotDeletedFilter(t: T): Rep[Option[Boolean]]
```

Comment:
Why `Option` as the return type of the filter function?

Reply:
i think for some reason it handles both Rep[Boolean] and Rep[Option[Boolean]]

Reply:
maybe type erasure something?